pkg/extension/sumologicextension/credentials/credentialsstore_localfs_test.go

```diff
@@ -74,3 +74,17 @@ func TestCredentialsStoreLocalFs(t *testing.T) {
 	)
 	require.EqualValues(t, fileCounter, 0)
 }
+
+func TestCredentialsStoreEnsureWritable(t *testing.T) {
+	dir := filepath.Join(t.TempDir(), "store")
+	expectedFileMode := fs.FileMode(0700)
+	err := os.Mkdir(dir, 0400)
+	require.NoError(t, err)
+
+	_, err = NewLocalFsStore(WithCredentialsDirectory(dir), WithLogger(zap.NewNop()))
+	require.NoError(t, err)
+
+	stat, err := os.Stat(dir)
```

Comment:
doesn't `ensureDir` already create the directory with the right permissions? why do we have `os.Stat` here again...